ion-transformation-storage-api/src/main/java/com/connexta/transformation/storage/api/Intel.java

```diff
@@ -0,0 +1,32 @@
+/**
+ * Copyright (c) Connexta
+ *
+ * <p>This is free software: you can redistribute it and/or modify it under the terms of the GNU
+ * Lesser General Public License as published by the Free Software Foundation, either version 3 of
+ * the License, or any later version.
+ *
+ * <p>This program is distributed in the hope that it will be useful, but WITHOUT ANY WARRANTY;
+ * without even the implied warranty of MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+ * GNU Lesser General Public License for more details. A copy of the GNU Lesser General Public
+ * License is distributed along with this program and can be found at
+ * <http://www.gnu.org/licenses/lgpl.html>.
+ */
+package com.connexta.transformation.storage.api;
+
+public interface Intel extends IntelInfo {
```

Comment:
❗️ I really don't think we should design this interface with the notion that the whole thing is loaded in memory by having manipulate objects. First that would require the implementation to know how to persist each type and second it would force us to load the whole thing in memory. We need the `save()` operation to expect and the `load()` operation to return an `InputStream` such that we are left with the possibility of streaming from one place to the other without necessarily loading all of it in memory.